googlemaps/places.py

```diff
@@ -0,0 +1,203 @@
+    if language:
+        params["language"] = language
+    if min_price:
+        params["minprice"] = min_price
+    if max_price:
+        params["maxprice"] = max_price
+    if open_now:
+        params["opennow"] = "true"
+    if page_token:
+        params["pagetoken"] = page_token
+
+    return client._get("/maps/api/place/textsearch/json", params)
+
+
+def place(client, place_id, language=None):
+    """
+    Comprehensive details for an individual place.
+
+    :param place_id: A textual identifier that uniquely identifies a place,
+                     returned from a Places search.
+    :type place_id: string
+
+    :param language: The language in which to return results.
+    :type langauge: string
+
+    :rtype: result dict with the following keys:
+            status: status code
+            result: dict containing place details
+            html_attributions: set of attributions which must be displayed
+
+    """
+    params = {"placeid": place_id}
+    if language:
+        params["language"] = language
+    return client._get("/maps/api/place/details/json", params)
+
+
+def photo(client, photo_reference, max_width=None, max_height=None):
+    """
+    Downloads a photo from the Places API.
+
+    :param photo_reference: A string identifier that uniquely identifies a
+      photo, as provided by either a Places search or Places detail request.
+    :type photo_reference: string
+
+    :param max_width: Specifies the maximum desired width, in pixels.
+    :type max_width: int
+
+    :param max_height: Specifies the maximum desired height, in pixels.
+    :type max_height: int
+
+    :rtype: iterator containing the raw image data, which typically can be
+      used to save an image file locally, eg:
+
+    ```
+    f = open(local_filename, 'wb')
+    for chunk in client.photo(photo_reference, max_width=100):
+        if chunk:
+            f.write(chunk)
+    f.close()
+    ```
+
+    """
+
+    if not (max_width or max_height):
+        raise ValueError("a max_width or max_height arg is required")
+
+    params = {"photoreference": photo_reference}
+
+    if max_width:
+        params["maxwidth"] = max_width
+    if max_height:
+        params["maxheight"] = max_height
+
+    # "extract_body" and "stream" args here are used to return an iterable
+    # response containing the image file data, rather than converting from
+    # json.
+    response = client._get("/maps/api/place/photo", params,
+                           extract_body=lambda response: response,
+                           requests_kwargs={"stream": True})
+    return response.iter_content()
+
+
+def autocomplete(client, input_text, offset=None, location=None,
+                       radius=None, language=None):
+    """
+    Returns Place predictions given a textual search query, such as
+    "pizza near New York", and optional geographic bounds.
+
+    :param input_text: The text query on which to search.
+    :type input_text: string
```

Comment:
The API param is called input but that clashes with Python's built-in function. Actually I think pep8 suggest `input_` in this case but I think it looks ugly, what do you think?